cpp/src/rolling/rolling_detail.cuh

```diff
@@ -33,6 +33,7 @@
 #include <cudf/detail/gather.hpp>
 #include <cudf/detail/groupby/sort_helper.hpp>
 #include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/detail/unary.hpp>
```

Comment:
```suggestion
```
Is this include needed? 
Actually, there are alot of includes here for a header file. Could you make sure all of them are necessary for the code in just this file? 

Some candidates for removal: `copying.hpp`, `valid_if.cuh`, `nvtx/ranges.hpp`, `strings/detail/utilities.cuh`, `thrust/pair.h`, `thrust/transform.h`, `thrust/execution_policy.h`, `thrust/detail/execution_policy.h`, `thrust/binary_search.h`

This may beyond the scope of this PR so maybe at least check the ones added by this PR are still required.

Reply:
It's needed for calling `SQRT` on the result of var to compute standard deviation.

Reply:
I can further rid of `cudf/rolling.hpp`, doing so would unearth an indirect include issue:

https://github.com/rapidsai/cudf/blob/1935a8a9de87152e70b7930c911e0a44da0560cc/cpp/src/rolling/rolling.cu#L23
actually depends on one of the declarations in that file, but was indirectly referenced via `rolling.hpp` - included in `rolling_detail.cuh`. Moving the function to after the referenced signature solves the issue. (included in this PR).